src/Data/Bin/Bin0.agda

```diff
@@ -0,0 +1,686 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Arithmetic of binary represented natural numbers. Initial part.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Bin.Bin0 where
+
+open import Algebra using (Magma; Semigroup; Monoid; CommutativeMonoid)
+open import Algebra.FunctionProperties using (Op₂; Associative; Commutative)
+open import Algebra.Structures         using (IsMagma; IsSemigroup; IsMonoid;
+                                                          IsCommutativeMonoid)
+open import Data.Nat using (ℕ)
+                     renaming (suc to 1+_; pred to pred'; _+_ to _+'_;
+                               _*_ to _*'_; _∸_ to _∸'_)
+     --
+     -- Here renaming is applied in order to distinguish the
+     -- corresponding operators from their Bin versions.
+     --
+open import Data.Nat.Properties as NatP using (2m≢1+2n)
```

Comment:
See Data.Integer.Properties for how we usually handle this instead of renaming everything.